python_modules/libraries/dagster-azure/dagster_azure_tests/adls2_tests/test_adls2_file_manager.py

```diff
@@ -180,6 +185,83 @@ def test_op(context):
     assert did_it_run["it_ran"]
 
 
+@mock.patch("dagster_azure.adls2.resources.ADLS2Resource")
+@mock.patch("dagster_azure.adls2.resources.ADLS2FileManager")
+def test_adls_file_manager_resource_cloud_type(MockADLS2FileManager, MockADLS2Resource):
+    did_it_run = dict(it_ran=False)
+
+    resource_config = {
+        "storage_account": "some-storage-account",
+        "credential": {
+            "key": "some-key",
+        },
+        "adls2_file_system": "some-file-system",
+        "adls2_prefix": "some-prefix",
+        "cloud_type": "government",
+    }
+
+    @op(required_resource_keys={"file_manager"})
+    def test_op(context):
+        # test that we got back a ADLS2FileManager
+        assert context.resources.file_manager == MockADLS2FileManager.return_value
+
+        # make sure the file manager was initalized with the config we are supplying
+        MockADLS2FileManager.assert_called_once_with(
+            adls2_client=MockADLS2Resource.return_value.adls2_client,
+            file_system=resource_config["adls2_file_system"],
+            prefix=resource_config["adls2_prefix"],
+        )
+        MockADLS2Resource.assert_called_once_with(
+            storage_account=resource_config["storage_account"],
+            credential=ADLS2Key(key=resource_config["credential"]["key"]),
+            cloud_type=resource_config["cloud_type"],
+        )
+
+        did_it_run["it_ran"] = True
+
+    context = build_op_context(
+        resources={"file_manager": configured(adls2_file_manager)(resource_config)},
+    )
+    test_op(context)
+    assert did_it_run["it_ran"]
+
+
+@pytest.mark.parametrize("cloud_type", ["government", "public", None])
+def test_adls_file_handle_adls2_path(cloud_type):
+    resource_config = {
+        "storage_account": "some-storage-account",
+        "credential": {
+            "key": "some-key",
+        },
+        "adls2_file_system": "some-file-system",
+        "adls2_prefix": "some-prefix",
+    }
+    if cloud_type:
+        resource_config["cloud_type"] = cloud_type
+
+    context = build_op_context(
+        resources={"file_manager": configured(adls2_file_manager)(resource_config)},
+    )
+    adls_file_manager = context.resources.file_manager
+
+    with mock.patch.object(
+        adls_file_manager,
+        "write_data",
+        return_value=ADLS2FileHandle(
+            resource_config["storage_account"],
+            resource_config["adls2_file_system"],
+            "some-key",
+            adls_file_manager.get_client().primary_endpoint,
```

Comment:
When I substitute `adls_file_manager._client.primary_endpoint` and run `make ruff`, it throws this error: `SLF001 Private member accessed: _client`. Is there a way to avoid this error without having a `get_client()` function?

Reply:
edited to use `._client` with `# noqa`